### ISSUE TITLE
NFR: Support MySQL 5.7 JSON data type.

### ISSUE THREAD

**Issue by hdogan** [tom--/dynamic-ar#13]
https://dev.mysql.com/doc/refman/5.7/en/json.html

**Comment by tom--** [comment]
Yes. A thousand times yes. I just don't think I'll have a chance to work on this in the near future.

If you or anyone else is interested to contribute, start with the work that @nineinchnick did in the pgsql-encoder branch. He refactored, extracting the DB-specific encoder parts, and provided a rough 1st version of a pgsql encoder. There should be considerable overlap between the pgsql and mysql parts, since they are both JSON oriented.
